src/restapi/fComplete.ts

```diff
@@ -53,11 +53,12 @@ export async function fComplete(req: Request, res: Response): Promise<void> {
 
 // createContextFromState constructs an AppContext from state. fComplete needs
 // access to the PluginKVStore and retrieves the token and url from state
-function createContextFromState(botToken: string, url: string): AppContext {
-    const context: AppContext = {
+function createContextFromState(botToken: string, url: string): AppContextWithBot {
+    const context: AppContextWithBot = {
+        app_id: '',
         bot_access_token: botToken,
         mattermost_site_url: url,
-    };
+    } as AppContextWithBot;
```

Comment:
Do we need the `as`?

Reply:
Yes. it is needed.  Without, it, the variable `context` will be an object of unknown type